### PR TITLE
Relocate json files

### PR DESCRIPTION
**Pull request by horozal**
Moved the `jsonfiles` folder into `src/main/resources` and updated the source code accordingly for reading from relocated json files.

### CODE REVIEW

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 1]
```diff
@@ -9,73 +9,76 @@
...
 public class PatternSelectionController {
 	
 	EnumMap<ArchitecturalPatterns, Integer> patternValues = new EnumMap<>(ArchitecturalPatterns.class);
 	
+	final URL url = this.getClass().getResource("/jsonfiles/surveyEvaluation.json");
+	
 	@PostMapping("/evaluation")
 	public String evaluateSurveyInput(@RequestBody String input){
+		
+		if (input == null) {
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Good to know! Removed it.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 2]
```diff
@@ -9,73 +9,76 @@
...
+			throw new NullPointerException("Survey input is empty.");
+			
+		} else {
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Removed.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 3]
```diff
@@ -9,73 +9,76 @@
...
 
 @RestController
 public class PatternSelectionController {
 	
 	EnumMap<ArchitecturalPatterns, Integer> patternValues = new EnumMap<>(ArchitecturalPatterns.class);
 	
+	final URL url = this.getClass().getResource("/jsonfiles/surveyEvaluation.json");
+	
 	@PostMapping("/evaluation")
 	public String evaluateSurveyInput(@RequestBody String input){
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Good to know! Updated accordingly.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 4]
```diff
@@ -9,73 +9,76 @@
...
+import java.util.Objects;
 
 @RestController
 public class PatternSelectionController {
 	
 	EnumMap<ArchitecturalPatterns, Integer> patternValues = new EnumMap<>(ArchitecturalPatterns.class);
 	
+	final URL url = this.getClass().getResource("/jsonfiles/surveyEvaluation.json");
+	
 	@PostMapping("/evaluation")
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Updated accordingly.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/SurveyController.java [hunk 5]
```diff
@@ -3,38 +3,38 @@
...
 
 @RestController
 public class SurveyController {
-	
-	private final String SURVEY_JSON_FILE_PATH =
-			"jsonfiles/survey.json";
+		
+	final URL url = this.getClass().getResource("/jsonfiles/survey.json");
 	
 	ObjectMapper mapper = new ObjectMapper();
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/SurveyController.java:21]: Done.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/SurveyController.java [hunk 6]
```diff
@@ -3,38 +3,38 @@
...
-		 String surveyJsonStr = null;
+		
+		String surveyJsonStr = null;
 		 
-		 try {
-			 surveyJsonStr = Files.readString(filePath);
-			 //jsonNode = mapper.readTree(surveyJsonStr);			 
+		try {
+			
+		 final Path filePath = Path.of(Objects.requireNonNull(url).toURI());
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/SurveyController.java]: Done.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 7]
```diff
@@ -9,73 +9,76 @@
...
-				String jsonStr = Files.readString(filePath);
-				JsonNode jsonNode = mapper.readValue(jsonStr, JsonNode.class);			
-				
 				//Iterate over survey question IDs
-				ListIterator<String> itr = ids.listIterator();			    			    
+				ListIterator<String> itr = ids.listIterator();	
+			
 				while (itr.hasNext()) {		
 					String next = itr.next();
 					valuesJsonNode = jsonNode.get(next).get(0);
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Done.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 8]
```diff
@@ -9,73 +9,76 @@
...
-			} else {
 				//Get survey question IDs from into string list
 				List<String> ids = mapper.readValue(input, new TypeReference<List<String>>(){});
 				
-				String jsonStr = Files.readString(filePath);
-				JsonNode jsonNode = mapper.readValue(jsonStr, JsonNode.class);			
-				
 				//Iterate over survey question IDs
-				ListIterator<String> itr = ids.listIterator();			    			    
+				ListIterator<String> itr = ids.listIterator();
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Yes, it does achieve the same. Replaced.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 9]
```diff
@@ -9,73 +9,76 @@
...
+			//Initialize 
+			patternValues.put(ArchitecturalPatterns.LAYERED, 0);
+			patternValues.put(ArchitecturalPatterns.EVENT_DRIVEN, 0);
+			patternValues.put(ArchitecturalPatterns.MICROKERNEL, 0);
+			patternValues.put(ArchitecturalPatterns.MICROSERVICES, 0);
+			patternValues.put(ArchitecturalPatterns.SERVICE_ORIENTED, 0);
+			patternValues.put(ArchitecturalPatterns.SPACE_BASED, 0);
 		
-		List<ArchitecturalPatterns> patternList = Arrays.asList(ArchitecturalPatterns.values());
+			List<ArchitecturalPatterns> patternList = Arrays.asList(ArchitecturalPatterns.values());
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Done.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/SurveyController.java [hunk 10]
```diff
@@ -3,38 +3,38 @@
...
+		 surveyJsonStr = Files.readString(filePath);
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/SurveyController.java]: Yes. Done.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 11]
```diff
@@ -9,73 +9,76 @@
...
-		
-		JsonNode valuesJsonNode = null;
-
-		try {	
+			try {	
+			
+				//Read evaluation values from JSON file
+				final Path filePath = Path.of(Objects.requireNonNull(url).toURI());
+				String jsonStr = Files.readString(filePath);
+				JsonNode jsonNode = mapper.readValue(jsonStr, JsonNode.class);
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Yes. Done.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 12]
```diff
@@ -9,73 +9,76 @@
...
+				//Read evaluation values from JSON file
+				final Path filePath = Path.of(Objects.requireNonNull(url).toURI());
+				String jsonStr = Files.readString(filePath);
+				JsonNode jsonNode = mapper.readValue(jsonStr, JsonNode.class);	
 			
-			if (input == null) {
-				throw new NullPointerException("Survey input is empty.");
-			} else {
 				//Get survey question IDs from into string list
 				List<String> ids = mapper.readValue(input, new TypeReference<List<String>>(){});
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: I now manually check if `jsonNode.get(id)` is null.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 13]
```diff
@@ -14,18 +14,38 @@
...
+
+		try {
+
+			// Read evaluation values from JSON file
+			final Path filePath = Path.of(Objects.requireNonNull(url).toURI());
+			jsonStr = Files.readString(filePath);
+			node = mapper.readValue(jsonStr, JsonNode.class);
+
+		} catch (URISyntaxException | IOException e) {
+			e.printStackTrace();
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java:42]: @philipreimer Do you have a preference between SLF4J and Log4j 2?

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 14]
```diff
@@ -14,18 +14,38 @@
...
-import java.util.Arrays;
 import java.util.EnumMap;
 import java.util.List;
-import java.util.ListIterator;
 import java.util.Objects;
 
 @RestController
 public class PatternSelectionController {
 
 	EnumMap<ArchitecturalPatterns, Integer> patternValues = new EnumMap<>(ArchitecturalPatterns.class);
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Done.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 15]
```diff
@@ -14,18 +14,38 @@
...
 
-	final URL url = this.getClass().getResource("/jsonfiles/surveyEvaluation.json");
+	public JsonNode readSurveyEvaluation() {
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Made it `private`. Problem with making it also `static`, cannot make a static reference to the non-static field `url`

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java (hunk 13 above)
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: OK, switched to SLF4J.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 16]
```diff
@@ -14,18 +14,38 @@
...
+		String jsonStr = null;
+		JsonNode node = null;
+		ObjectMapper mapper = new ObjectMapper();
+
+		try {
+
+			// Read evaluation values from JSON file
+			final Path filePath = Path.of(Objects.requireNonNull(url).toURI());
+			jsonStr = Files.readString(filePath);
+			node = mapper.readValue(jsonStr, JsonNode.class);
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Done.

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java [hunk 17]
```diff
@@ -14,18 +14,38 @@
...
+
+		} catch (URISyntaxException | IOException e) {
+			e.printStackTrace();
+		}
+
+		return node;
+	}
+
+	JsonNode jsonNode = readSurveyEvaluation();
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/PatternSelectionController.java]: Added `if (surveyEvaluationNode == null) throw new NullPointerException("Evaluation values are not being referenced.");`

File: src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/SurveyController.java [hunk 18]
```diff
@@ -16,15 +14,9 @@
...
 
-	final URL url = this.getClass().getResource("/jsonfiles/survey.json");
-
-	ObjectMapper mapper = new ObjectMapper();
-
-	@GetMapping(value = "/survey", produces = MediaType.APPLICATION_JSON_VALUE)
-
-	public String getSurvey() {
-
+	public String readSurvey() {
```
[the remark this answers is not in the document]
Comment (horozal) [reply src/main/java/org/eclipse/opensmartclide/architecturalpatterns/selection/SurveyController.java]: Done in the same manner as for `PatternSelectionController`

**Review by schlotze — APPROVED** [review]

**Pull request merged**